Lib/test/test_unpack.py

```diff
@@ -167,5 +220,49 @@ def test_extended_oparg_not_ignored(self):
             y = unpack_400(range(400))
             self.assertEqual(y, 399)
 
+    def test_exception_context_when_len_fails(self):
+        """When `__len__()` raises an Exception, preserve exception context"""
+        code = textwrap.dedent(
+            """
+            class CustomSeq:
+                def __len__(self):
+                    raise RuntimeError
+
+                def __getitem__(self, i):
+                    return i*2
+
+            x, y, z = CustomSeq()
+            """
+        )
+        with self.assertRaises(UnpackError) as cm:
+            exec(code)
```

Comment:
```suggestion
		class LenRaisesException:
			def __len__(self):
				raise RuntimeError
            def __getitem__(self, i):
                return i*2

        with self.assertRaises(UnpackError) as cm:
            x, y, z = LenRaisesException()
```

This should work as well I think. Personally, I would, in this case however expect a RuntimeError though. This is what currently happens when `__getitem__` fails.